substrate/client/basic-authorship/src/basic_authorship.rs

```diff
@@ -565,20 +565,22 @@ where
 
 		if log::log_enabled!(log::Level::Info) {
 			info!(
-				"🎁 Prepared block for proposing at {} ({} ms) [hash: {:?}; parent_hash: {}; extrinsics_count: {}",
+				"🎁 Prepared block for proposing at {} ({} ms) hash: {:?}; parent_hash: {}; end: {:?}; extrinsics_count: {};",
 				block.header().number(),
 				block_took.as_millis(),
 				<Block as BlockT>::Hash::from(block.header().hash()),
 				block.header().parent_hash(),
+				end_reason,
```

Comment:
```suggestion
				"🎁 Prepared block for proposing at {} ({} ms) hash: {:?}; parent_hash: {}; extrinsics_count: {};",
				block.header().number(),
				block_took.as_millis(),
				<Block as BlockT>::Hash::from(block.header().hash()),
				block.header().parent_hash(),
```

No need to add log this in info.

Reply:
I need it here, this is why I added it. I need to know why block builder is hitting limits. And we may have it "for free" in this line.

I am now confused, it is OK to merge it with end reason or not?

Reply:
Why do you need it there? You could just enable trace logging.

Reply:
Also we have metrics for this.

Reply:
I think its useful to have, with tracing it will spam all tx again.

Reply:
I don't want to enable trace, because it is way too spammy. With 7k txs in block it may affect the block time building. And for sure it does not help with collecting logs in grafana/loki - it sometimes breaks with too many logs.

I can add another "if branch" here for debug, but I think it will more confusing when reasoning about what to enable.

Reply:
It is not one network, instead of having important info in log I always need to restart the node whiat is a cost (time). having this log allows very quick assessment of what is happening in block builder. 

and logs are not only for operator. after all why does operator need this hash?

I would not propose it if I could live w/o this. And finally @skunert says he finds it useful too.